source/slang/slang-linkable.cpp

```diff
@@ -855,6 +903,10 @@ Expr* ComponentType::findDeclFromString(String const& name, DiagnosticSink* sink
     {
         result = checkedExpr;
     }
+    else if (auto overloadExpr2 = as<OverloadedExpr2>(checkedExpr))
+    {
+        result = convertOverloadExpr2(overloadExpr2, astBuilder);
+    }
```

Comment:
It might be good to define a subroutine along the lines of:

    Expr* maybeSimplifyExprForReflectionAPIUsage(Expr* originalExpr, ASTBuilder* astBuilder);

That function would be a catch-all to try to simplify the given `originalExpr` in ways that may make it more useful as a result to expose via the reflection API, and return either a simplified expression or the original expression.

That subroutine could handle the conditional check for `OverloadedExpr2`, so that this and the other use site would just become calls like:

    result = maybeSimplifyExprForReflectionAPIUsage(result, astBuilder);

That refactor would keep the messy details out of the flow here, and also make it more clear *why* this logic is being invoked in the first place.